specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2021-09-01-privatepreview/managedapplication.json

```diff
@@ -0,0 +1,2729 @@
+        }
+      ],
+      "x-ms-discriminator-value": "AddRemoveIncrementalNamedPartition",
+      "required": [
+        "minPartitionCount",
+        "maxPartitionCount",
+        "scaleIncrement"
+      ],
+      "properties": {
+        "minPartitionCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Minimum number of named partitions of the service."
+        },
+        "maxPartitionCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Maximum number of named partitions of the service."
+        },
+        "scaleIncrement": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of instances to add or remove during a scaling operation."
+        }
+      }
+    },
+    "ApplicationHealthPolicy": {
+      "type": "object",
+      "description": "Defines a health policy used to evaluate the health of an application or one of its children entities.\n",
+      "required": [
+        "considerWarningAsError",
+        "maxPercentUnhealthyDeployedApplications"
+      ],
+      "properties": {
+        "considerWarningAsError": {
+          "type": "boolean",
+          "description": "Indicates whether warnings are treated with the same severity as errors."
+        },
+        "maxPercentUnhealthyDeployedApplications": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The maximum allowed percentage of unhealthy deployed applications. Allowed values are Byte values from zero to 100.\nThe percentage represents the maximum tolerated percentage of deployed applications that can be unhealthy before the application is considered in error.\nThis is calculated by dividing the number of unhealthy deployed applications over the number of nodes where the application is currently deployed on in the cluster.\nThe computation rounds up to tolerate one failure on small numbers of nodes. Default percentage is zero.\n"
+        },
+        "defaultServiceTypeHealthPolicy": {
+          "$ref": "#/definitions/ServiceTypeHealthPolicy",
+          "description": "The health policy used by default to evaluate the health of a service type."
+        },
+        "serviceTypeHealthPolicyMap": {
+          "$ref": "#/definitions/ServiceTypeHealthPolicyMap",
+          "description": "The map with service type health policy per service type name. The map is empty by default."
+        }
+      }
+    },
+    "ApplicationParameterList": {
+      "type": "object",
+      "description": "List of application parameters with overridden values from their default values specified in the application manifest.",
+      "additionalProperties": {
+        "type": "string"
+      }
+    },
+    "ApplicationResource": {
+      "type": "object",
+      "description": "The application resource.",
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/ManagedIdentity"
```

Comment:
Can you use the common types v3 definition for ManagedServiceIdentity here instead? https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json

Reply:
@TimLovellSmith we don't want to do that. our model is a wrapper on top of this and if that changes we don't want to have a breaking change. and we could also add some more things in the future. So we prefer to keep it as is

Reply:
@a-santamaria No breaking change will be forced on you. The specification in common types v3 won't ever have a breaking change. If there is ever a breaking change, the change will be as a new version, common types v4, and you would only get those changes when someone intentionally updates your swagger to use the new def, because they want the breaking change.